### PR TITLE
feat: add artifact and typechain exports

### PR DESCRIPTION
**Pull request by mrice32**
This adds typechain and an artifact json export that details the artifact locations. This will allow importers of this package to run post-processing on the contracts and their hardhat artifacts.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,14 +20,19 @@
...
     "@uma/contracts-node": "^0.1.15",
     "@uniswap/v3-core": "^1.0.0-rc.2",
     "arb-ts": "^1.0.2",
     "decimal.js": "^10.3.1",
     "ethers": "^5.5.2",
     "glob": "^7.2.0",
-    "hardhat": "^2.6.8"
+    "hardhat": "^2.6.8",
+    "lodash.uniqby": "^4.7.0",
+    "typechain": "^6.0.0"
```
Comment (chrismaree) [inline package.json]: we now have type chain in common, contracts-node, contracts-frontend and core. do all these locations need it given this change?

Comment (mrice32) [reply package.json]: It's a good point. Will double check if we can drop it everywhere outside of common. I think I've tried this before and had issues. I'll get to the bottom of it.

Comment (mrice32) [reply package.json]: So typechain isn't a dependency of common, and everywhere else it's a devDependency. This means that we don't "inherit" the installation from anywhere. I tried removing it and got an error message saying it was necessary.

We could technically make it a regular dependency of common if we want, but that change would need to be made in protocol. I'm also not sure that users of common need it unless they specifically override the config to force typechain types to be compiled.

Thoughts?

Comment (chrismaree) [reply package.json]: ye, that seems fine to leave it in that case. 

as a related point, do we know of any tools that can go through our lerna packages and look for ways to simplifying the dependency tree? i.e looking for packages we import and dont need and perhaps moving some stuff around to have fewer dependencies in each package?

Comment (mrice32) [reply package.json]: I don't! This would be useful, though! Would be interested if you have a suggestion.

**Review by nicholaspai — APPROVED** [review]
lgtm

**Pull request merged**